server/src/main/java/org/elasticsearch/snapshots/RestoreService.java

```diff
@@ -269,7 +269,7 @@ public ClusterState execute(ClusterState currentState) {
                                 if (currentIndexMetaData == null) {
                                     // Index doesn't exist - create it and start recovery
                                     // Make sure that the index we are about to create has a validate name
-                                    MetaDataCreateIndexService.validateIndexName(renamedIndexName, currentState);
+                                    MetaDataCreateIndexService.validateIndexName(renamedIndexName, currentState, Collections.emptySet());
```

Comment:
I feel like this should have the actual value for the allowed system indices and it looks like this code has access to the instance of the MetaDataCreateIndexService so maybe we could have a non-static method

Reply:
Agreed - for some reason I thought it was more difficult to make `MetaDataCreateIndexService.validateIndexName` non-static but everywhere that uses it currently already has a `MetaDataCreateIndexService` instance anyway - it looks like we can just make this method non-static.